server/src/main/java/io/crate/expression/operator/any/AnyOperator.java

```diff
@@ -167,6 +167,19 @@ public Query toQuery(Function function, Context context) {
         } else if (probe instanceof Reference ref && candidates instanceof Literal<?> literal) {
             return refMatchesAnyArrayLiteral(function, ref, literal, context);
         } else {
+            // TODO: why do we use generic query ?
```

Comment:
because you are comparing 2 columns of the table, not a column with a literal, so there is not way you can use lucene queries since to apply the filter you need to get for each row the values of the 2 columns and then apply the generic function filter query.